test/e2e/fifo_queue.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
```

Comment:
@bprashanth I looked at converting to use the client/cache/fifo.go but as far as I can see no one actually uses that implementation.  There is a DeltaFIFO which is used, but it's a separate implementation.  I was able to pretty easily convert to use the cache implementation, but in its current form I don't see how it can work.  That FIFO is really built such that Add and Pop work as a FIFO would, but List does not.  List creates a new list from the hash table instead of from the internal queue so you can lose ordering in the output of List.  Since List isn't usable I don't see any way to non-destructively go through the items in the FIFO in the correct order as would be needed by the rc pod submission tracking portion of the test.

Am I missing something?